recaf-core/build.gradle

```diff
@@ -38,7 +38,8 @@ dependencies {
     api(libs.openrewrite)
     api(libs.regex)
     api(libs.bundles.jasm)
-    api(libs.vineflower)
+//    api(libs.vineflower)
+    api( project(':recaf-relocation'))
```

Comment:
I know that you put in a fair amount of effort to get these decompilers to all work with one another, but I am rather against having such things like this relocation module in the main repository.

Additionally, the amount of differences between these decompilers is rather small since they are all forked off one-another. 

IMO these alternative flowers should be offered as plugins which are independently shading their respective flower code into the plugin jars. Decompiler implementations and config containers can be registed on the fly so there's no concern with lessened capability by having them offered separately.

Plus, given some of our user interactions I have a strong feeling that having nearly the same decompiler three times would be confusing to them.

Reply:
> Additionally, the amount of differences between these decompilers is rather small since they are all forked off one-another.

Although I know that they are forked from each other, I cannot agree that their differences are small. In fact, after years of evolution, they are now very different in terms of interface form, number of controllable parameters, and compilation results.
In terms of the number of parameters, vineflower: 64, fernflower: 45, forgeflower: 47.
The following three figures are the decompilation results of the three flower decompilers for `software.coley.instrument.io.ByteBufferSanitizer` in `%APPDATA%/Recaf/agent/agent-1.4.2.jar` in the Recaf running directory. The source code of the corresponding class is located in [ByteBufferSanitizer](https://github.com/Col-E/InstrumentationServer/blob/master/src/main/java/software/coley/instrument/io/ByteBufferSanitizer.java).  Among them, only fernflower obtained the equivalent result to the source code.
![Vineflower](https://github.com/user-attachments/assets/759e5960-d736-458f-93ff-44169438c076)
![Fernflower](https://github.com/user-attachments/assets/4d7941bf-cc0c-4741-98d2-7576f217a106)
![Forgeflower](https://github.com/user-attachments/assets/b3a72315-8461-4c0f-aa19-9080250a5f43)

When decompiling some internal enumeration classes, vineflower reports an error, while fernflower and forgeflower can get the decompiled results. Below are the decompiled results of the three flower class decompilers for `jadx.api.JadxArgs$RenameEnum` in [jadx-core-1.5.0.jar](https://repo1.maven.org/maven2/io/github/skylot/jadx-core/1.5.0/jadx-core-1.5.0.jar)

![Vineflower-1](https://github.com/user-attachments/assets/d2c6c497-21ee-49b8-83a0-11c0499df1a8)
![Fernflower-1](https://github.com/user-attachments/assets/36577aa0-cb29-489c-af18-4a6f0a8ea845)
![Forgeflower-1](https://github.com/user-attachments/assets/698841ca-a10f-4491-a0cc-b2eb17710eab)





> IMO these alternative flowers should be offered as plugins which are independently shading their respective flower code into the plugin jars. Decompiler implementations and config containers can be registed on the fly so there's no concern with lessened capability by having them offered separately.

Maybe this is one of the possible solutions, but I don't know how to create a plug-in for Recaf yet. Is there any sample for reference?

> Plus, given some of our user interactions I have a strong feeling that having nearly the same decompiler three times would be confusing to them.

For ordinary users, they mostly use the default CFR decompiler and use the default configuration. For professional users, they should know clearly which compiler they want to use and what configuration parameters they want to use. Maybe I need to further optimize the configuration name to make it more clearly match the parameter name in the corresponding decompiler official documentation.